components/resources/quizzes/shared/SkillSelections.tsx

```diff
@@ -13,20 +18,49 @@ type SkillSelectionsProps = {
 
 const SkillSelections: React.FC<SkillSelectionsProps> = ({
   selections,
+  selected,
+  setSelected,
+  setQuizResponses,
+  currentStage,
   onNextClick,
   onBackClick,
   progress,
   title,
   body,
 }) => {
-  const [selected, setSelected] = useState<string[]>([]);
+  const currentSelections = selected[currentStage] || [];
+
   const handleSelection = (selection: string) => {
-    if (selected.includes(selection)) {
-      setSelected(selected.filter((s) => s !== selection));
+    const selectionIndex = currentSelections.indexOf(selection);
+    if (selectionIndex !== -1) {
+      setSelected((prev) => {
+        const updatedSelections = [...prev];
+        updatedSelections[currentStage] = currentSelections.filter(
+          (s, index) => index !== selectionIndex
+        );
+        return updatedSelections;
+      });
     } else {
-      setSelected([...selected, selection]);
+      setSelected((prev) => {
+        const updatedSelections = [...prev];
+        updatedSelections[currentStage] = [...currentSelections, selection];
+        return updatedSelections;
+      });
     }
   };
+
+  const handleStage = () => {
+    setQuizResponses((prev) => {
+      const updatedResponses = [...prev];
+      updatedResponses[currentStage - 1] = selected[currentStage];
+      return updatedResponses;
+    });
+    setSelected((prev) => {
```

Comment:
does this even change the state at all?

Reply:
Yeah, this is how I push the selections to the parent component. If that's what you are asking... Otherwise, I'm not sure??

Reply:
but you're just taking the previous state, and assigning it into a new variable and then re-assigning it to the same to the same state? 

isn't      `const updatedSelections = [...prev];`
the same thing as `const updatedSelections = prev` ?